src/main/java/uk/gov/hmcts/reform/cwrdapi/util/EmailValidator.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.hmcts.reform.cwrdapi.util;
+
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.ObjectUtils;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+@Slf4j
+@NoArgsConstructor
+/*
+EmailValidator IS EMAIL VALIDATION. THIS ALSO ALLOWED ONLY DOMAINS MENTIONED IN APPLICATION.YAML FILE
+@copyrights : hmcts
+ */
+public class EmailValidator implements ConstraintValidator<ValidateEmail, String> {
+
+    private String emailDomainList;
+
+    @Override
+    public void initialize(ValidateEmail validateEmail) {
+        this.emailDomainList = EmailDomainPropertyInitiator.emailDomains;
+    }
+
+    @Override
+    public boolean isValid(String emailAddress, ConstraintValidatorContext constraintValidatorContext) {
+        if (ObjectUtils.isEmpty(emailAddress) || ObjectUtils.isEmpty(this.emailDomainList)) {
+            log.error("Email id => '{}' or email.domainList => {} should not empty", emailAddress, emailDomainList);
```

Comment:
Can you please remove logging the email address as it is PII data?

Reply:
Ok